pubsub-mapped-api/src/main/java/com/google/pubsub/clients/producer/ExtendedConfig.java

```diff
@@ -0,0 +1,81 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+package com.google.pubsub.clients.producer;
+
+import java.util.Map;
+import com.google.common.annotations.VisibleForTesting;
+
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerConfigAdapter;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.AbstractConfig;
+import org.apache.kafka.common.config.ConfigDef.Type;
+import org.apache.kafka.common.config.ConfigDef.Range;
+import org.apache.kafka.common.config.ConfigDef.Importance;
+
+/**
+ * Provides the configurations for a Kafka Producer instance.
+ */
+@VisibleForTesting
+public class ExtendedConfig {
```

Comment:
Okay, but I think for the pubsub configs to be an inner class is the proper way - as it serves for nothing on its own -, isn't it?